src/main/java/org/jclouds/vcloud/director/v1_5/handlers/VcloudDirectorClientErrorRetryHandler.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jclouds.vcloud.director.v1_5.handlers;
+
+import static org.jclouds.http.HttpUtils.closeClientButKeepContentStream;
+
+import java.util.Set;
+
+import org.jclouds.http.HttpCommand;
+import org.jclouds.http.HttpResponse;
+import org.jclouds.http.annotation.ClientError;
+import org.jclouds.http.handlers.BackoffLimitedRetryHandler;
+import org.jclouds.vcloud.director.v1_5.domain.VcloudDirectorError;
+
+import com.google.inject.Inject;
+
+/**
+ * Handles Retryable responses with error codes in the 4xx range
+ */
+public class VcloudDirectorClientErrorRetryHandler extends BackoffLimitedRetryHandler {
+
+   private final VcloudDirectorUtils utils;
+   private final Set<String> retryableCodes;
+
+   @Inject
+   public VcloudDirectorClientErrorRetryHandler(VcloudDirectorUtils utils, 
```

Comment:
I think it is recommended to use `@Inject` with protected or private constructor.